clipboard/ios/Plugin/Clipboard.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import Capacitor
+
+public class Clipboard: NSObject {
+
+    enum ContentType: Int {
+        case string
+        case url
+        case image
+    }
+
+    struct ClipboardError: Error {
+        private let message: String
+
+        var localizedDescription: String {
+            return message
+        }
+
+        init(_ message: String) {
+            self.message = message
+        }
+    }
+
+    func write(content: String, ofType type: ContentType) -> Result<Void, Error> {
+        switch type {
+        case ContentType.string:
+            UIPasteboard.general.string = content
+            return .success(())
+        case ContentType.url:
+            if let url = URL(string: content) {
+                UIPasteboard.general.url = url
+                return .success(())
+            } else {
+                let errMsg = "Unable to form URL"
+                CAPLog.print(errMsg)
```

Comment:
It would be simpler to adopt the suggested enum, then return `ClipboardError.invalidURL` here. And let the calling class log the message to CAPLog.